java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestGenerationUtils.java

```diff
@@ -359,6 +373,60 @@ private static String getSuggestedMethodNameByAnnotation(String annotation) {
         }
     }
 
+    /**
+     * return modifier bit mask.
+     */
+    private static int getTestMethodModifiers(Map<String, IMethodBinding> methodsMap, TestKind kind,
+            String annotation, String methodName) {
+        int modifiers = Modifier.NONE;
+        // JUnit 4's test method must be public
+        if (kind == TestKind.JUnit) {
+            modifiers |= Modifier.PUBLIC;
+        }
+
+        // @BeforeClass and @AfterClass in JUnit 4 & 5 needs static modifier
+        if (needStaticModifier(kind, annotation)) {
+            modifiers |= Modifier.STATIC;
+        }
+
+        final IMethodBinding binding = methodsMap.get(methodName);
+        if (binding == null) {
+            return modifiers;
+        }
+
+        final int superModifiers = binding.getModifiers();
+        if (Modifier.isProtected(superModifiers)) {
+            modifiers |= Modifier.PROTECTED;
+        } else if (Modifier.isPublic(superModifiers)) {
+            modifiers |= Modifier.PUBLIC;
+        }
+
+        return modifiers;
+    }
+
+    private static Map<String, IMethodBinding> getMethodsBindings(ITypeBinding typeBinding) {
+        final Map<String, IMethodBinding> methods = new HashMap<>();
+        final IMethodBinding[] typeMethods = typeBinding.getDeclaredMethods();
+        for (final IMethodBinding methodBinding : typeMethods) {
+            methods.put(methodBinding.getName(), methodBinding);
+        }
+        ITypeBinding superClass = typeBinding.getSuperclass();
+        while (superClass != null) {
+            for (final IMethodBinding methodBinding : superClass.getDeclaredMethods()) {
+                if (methods.containsKey(methodBinding.getName())) {
+                    continue;
+                }
+
+                if (Modifier.isPrivate(methodBinding.getModifiers())) {
```

Comment:
https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html

Members with private or no modifier are not accessible to subclass. Maybe the condition could be

```java
if (Modifier.isProtected(...) || Modifier.isPublic(...)) {
  methods.put(methodBinding.getName(), methodBinding);
}
```